lib/src/rules/pub/no_http_urls.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/src/lint/pub.dart'; // ignore: implementation_imports
+
+import '../../analyzer.dart';
+
+const _desc = r"Don't use http urls.";
```

Comment:
The message isn't very helpful. Consider declaring a static field holding an instance of `LintCode` that has both a more complete problem message (something like "The 'http' scheme is not secure and shouldn't be used in a 'pubspec.yaml' file.") and a correction message (something like "Try using an 'https' scheme.".